ext/fg/js/document.js

```diff
@@ -92,7 +92,19 @@ function docRangeFromPoint(point) {
 
     if(imposter !== null) imposter.style.zIndex = -2147483646;
 
-    const rect = range.getClientRects()[0];
+    let rects;
+    try {
+        rects = range.getClientRects();
+    }
+    catch (e) {
+        return;
```

Comment:
What could cause `getClientRects` to throw here? Is it something that could be tested for explicitly as opposed to doing a blanket catch-all?

Reply:
Looks like it's ```TypeError: "range is undefined"``` coming from
```js
const range = document.caretRangeFromPoint(point.x, point.y);
```

I thought there was a different exception being thrown that was coming from ```getClientRects``` internally, but I guess not. Looks like it stems from the [fallback implementation](https://github.com/FooSoft/yomichan/pull/157/commits/33ec369eff87ace3e8b49c6461fa1b9a5d2a1202#diff-8dcb9a0ea4375b3bf7ed569620281840L79) of document.caretRangeFromPoint. Somehow that is being triggered in the extension on Firefox Developer despite the standard browser context having it defined.